src/test/java/de/tum/in/www1/artemis/util/CourseTestService.java

```diff
@@ -1524,6 +1526,66 @@ public void searchStudentsInCourse() throws Exception {
+        assertThat(resultTest5.stream().filter(UserPublicInfoDTO::getIsEditor).collect(Collectors.toList())).hasSize(numberOfEditors);
+        assertThat(resultTest5.stream().filter(UserPublicInfoDTO::getIsTeachingAssistant).collect(Collectors.toList())).hasSize(numberOfTutors);
+        assertThat(resultTest5.stream().filter(UserPublicInfoDTO::getIsInstructor).collect(Collectors.toList())).hasSize(numberOfInstructors);
+
+        // Test 6: Try to search for all tutors, editors and instructors with search term
+        var resultTest6 = searchUsersTest(course, List.of("tutors", "instructors"), Optional.of(userPrefix + "tutor"), numberOfTutors, true);
+        assertThat(resultTest6.stream().filter(UserPublicInfoDTO::getIsEditor).collect(Collectors.toList())).hasSize(0);
+        assertThat(resultTest6.stream().filter(UserPublicInfoDTO::getIsTeachingAssistant).collect(Collectors.toList())).hasSize(numberOfTutors);
+        assertThat(resultTest6.stream().filter(UserPublicInfoDTO::getIsInstructor).collect(Collectors.toList())).hasSize(0);
+
+        // Test 7: Try to search or all students, tutors, editors and instructors with a too short search term (at least 3 as students are included)
+        searchUsersTest(course, List.of("students", "tutors", "instructors"), Optional.of("tu"), 0, false);
+
+        // Test 8: Try to search or all students, tutors, editors and instructors with a long enough search term (at least 3 as students are included)
+        var resultTest8 = searchUsersTest(course, List.of("students", "tutors", "instructors"), Optional.of(userPrefix + "tutor"), numberOfTutors, true);
+        assertThat(resultTest8.stream().filter(UserPublicInfoDTO::getIsEditor).collect(Collectors.toList())).hasSize(0);
+        assertThat(resultTest8.stream().filter(UserPublicInfoDTO::getIsTeachingAssistant).collect(Collectors.toList())).hasSize(numberOfTutors);
+        assertThat(resultTest8.stream().filter(UserPublicInfoDTO::getIsInstructor).collect(Collectors.toList())).hasSize(0);
+        assertThat(resultTest8.stream().filter(UserPublicInfoDTO::getIsStudent).collect(Collectors.toList())).hasSize(0);
+    }
+
+    private List<UserPublicInfoDTO> searchUsersTest(Course course, List<String> roles, Optional<String> loginOrName, int expectedSize, boolean shouldPass) throws Exception {
+        MultiValueMap<String, String> queryParameter = new LinkedMultiValueMap<>();
+        queryParameter.add("loginOrName", loginOrName.orElse(""));
+        queryParameter.add("roles", String.join(",", roles));
+        var status = shouldPass ? HttpStatus.OK : HttpStatus.BAD_REQUEST;
+        var foundUsers = request.getList("/api/courses/" + course.getId() + "/users/search", status, UserPublicInfoDTO.class, queryParameter);
+        if (shouldPass) {
+            assertThat(foundUsers).hasSize(expectedSize);
+            return foundUsers;
+        }
+        else {
+            return emptyList();
```

Comment:
Shouldn't you verify that the rest call actually did not return any users in the list?

Reply:
I added an explicit null check. dd46be39bf33b33711a5bebb7b78036ddb7eba6d